src/OrchardCore/OrchardCore.Deployment.Abstractions/IDeploymentSourceOfT.cs

```diff
@@ -0,0 +1,14 @@
+namespace OrchardCore.Deployment;
+
+/// <summary>
+/// Contract for a deployment source.
+/// </summary>
+/// <typeparam name="TDeploymentStep">The deployment step type.</typeparam>
+public interface IDeploymentSource<TDeploymentStep>
```

Comment:
I don't think we need an interface for this. I think a base class is all you need.